### PR TITLE
Deathgasp changes for mutes

### LINKED ISSUE #19019: mute people can speak in their last whispers
## Description
as the name suggests, the newly added last words feature allows mute people to say anything they want as their last words

**Reproduction**
go crit as a mute person
say your last words

**Screenshots**
i forgot to get a screenshot

**Additional context**

### PR DESCRIPTION
**Pull request by Errant-4**
## About the PR

-Enshrines in stone the CentCom-given right of mute people to say their last words
-Mutes don't deathgasp when they die
-Mutes can't fake death

fixes #19019

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- tweak: Mutes no longer make a sound when they die, and can't fake their death while in crit. They can still summon up their remaining strength to whisper their first-last words.

### CODE REVIEW

**Comment by forgotmyotheraccount** [comment]
I-I was m-merely pretending...

**Comment by LankLTE** [comment]
Nanotrasen has the ability to cure mute people, and has given it to all of them, but also it fucking kills them if they use it

**Comment by Errant-4** [comment]
It's like spy drama cyanide teeth, except it's the mute cure (that also kills you)

**Comment by Emisse** [comment]
mute people shouldnt be able to death gasp

**Comment by Interrobang01** [comment]
This is good, fixing or keeping this doesn't really have an effect on gameplay either way, it's poetic and dramatic so probably good for roleplay, and _we don't talk about realism_.

**Comment by Partmedia** [comment]
I agree, mute people shouldn't be able to death gasp.

**Comment by Errant-4** [comment]
Disabled mute deathgasp as suggested

File: Content.Server/Mobs/CritMobActionsSystem.cs [hunk 1]
```diff
@@ -47,6 +52,12 @@ private void OnFakeDeath(EntityUid uid, MobStateActionsComponent component, Crit
         if (!_mobState.IsCritical(uid))
             return;
 
+        if (HasComp<MutedComponent>(uid))
```
Comment (Partmedia) [inline Content.Server/Mobs/CritMobActionsSystem.cs:55]: Please replace this with a call to ActionBlockerSystem's `CanSpeak()` method.

File: Content.Server/Mobs/CritMobActionsSystem.cs [hunk 2]
```diff
@@ -58,6 +69,7 @@ private void OnLastWords(EntityUid uid, MobStateActionsComponent component, Crit
         _quickDialog.OpenDialog(actor.PlayerSession, Loc.GetString("action-name-crit-last-words"), "",
             (string lastWords) =>
             {
+                // Intentionally does not check for muteness
```
Comment (Partmedia) [inline Content.Server/Mobs/CritMobActionsSystem.cs:72]: Please check for `CanSpeak()` here as well.

File: Content.Server/Mobs/DeathgaspSystem.cs [hunk 3]
```diff
@@ -33,6 +34,9 @@ public bool Deathgasp(EntityUid uid, DeathgaspComponent? component = null)
         if (!Resolve(uid, ref component, false))
             return false;
 
+        if (HasComp<MutedComponent>(uid))
```
Comment (Partmedia) [inline Content.Server/Mobs/DeathgaspSystem.cs:37]: And here.

**Review by Partmedia — DISMISSED** [review]
Thank you for revising your PR. There are a few places that are still missing the use of the ActionBlockerSystem.

File: Content.Server/Mobs/CritMobActionsSystem.cs (hunk 2 above)
Comment (Errant-4) [reply Content.Server/Mobs/CritMobActionsSystem.cs:72]: The entire start of this PR was the premise that this oversight should be embraced as a feature. Mirrorcult and Deltanedas were both supportive of the idea on the linked issue (19019)

File: Content.Server/Mobs/CritMobActionsSystem.cs (hunk 1 above)
Comment (Errant-4) [reply Content.Server/Mobs/CritMobActionsSystem.cs:55]: That won't work in this case because crit mobs can't speak

**Review by DrSmugleaf — APPROVED** [review]

File: Content.Server/Mobs/CritMobActionsSystem.cs (hunk 2 above)
Comment (DrSmugleaf) [reply Content.Server/Mobs/CritMobActionsSystem.cs:72]: For context https://github.com/space-wizards/space-station-14/issues/19019#issuecomment-1676025825

**Pull request merged**